Extracting text

Faktorealchik opened the pull request:
Added extracting text from epub file 
Added in example how to extract text (its extracting all text from book)
Small improvements on example project
Added weaks to delegates

Commit:
added weak on delegates
added extracting text by Spine
small improvements on Example
small improvements on Readme
added example of extracting all text

Commit:
removed team

f-meloni commented on EpubExtractor/Classes/Epub.swift:
```diff
@@ -109,5 +109,5 @@ public struct Epub {
     }
     
-    let epubContentParser: EPubContentParser
+    public let epubContentParser: EpubContentParser
```
Why do you need this to be public?

Faktorealchik replied:
Hello! :) 

To read contents (all book as a String) we need to access EpubContentParser 
in EpubContentParser I have placed all logic about parsing text as plain String 
so to access book as a String we need to get variable epubContentParser 

some PR more I will think about different solution, place it in, for example, Chapter, this will be more logical, so than we will able to remove this public

f-meloni replied:
Would be possible to make a public getter on the Epub for that?
Something like:

```swift
public var allBook: String {
  return epubContentParser.allBook
}
```

Faktorealchik replied:
Of course, I think about that when it was in the development, next commit will repair that

Faktorealchik replied:
currently removed spineitems from contentParser - because this is already in the Book, 
added methods to get all text - in Epub

f-meloni commented on EpubExtractor/Classes/EpubParser.swift:
```diff
@@ -62,9 +69,8 @@ final class EpubParser {
-        else {
-            return nil
-        }
+        
```
I would prefer to keep the else here, because makes it more safe
For example if I add another else and don't add a return by mistake, it will still compile
```swift
if let rootFileURLString = xmlDoc.root["rootfiles"].children.first?.attributes["full-path"] {
  return URL(fileURLWithPath: epubDirectoryURL.appendingPathComponent(rootFileURLString).path)	            
} else if X {
  _ = doX()
}

return nil
```
Compiles, but it wouldn't with the else

```swift
if let rootFileURLString = xmlDoc.root["rootfiles"].children.first?.attributes["full-path"] {
  return URL(fileURLWithPath: epubDirectoryURL.appendingPathComponent(rootFileURLString).path)	            
} else if X {
  _ = doX()
} else {
  return nil
}

Faktorealchik replied:
fixed in commit below:)

Faktorealchik replied:
I did it just in one reason - less code. 
When you do something with if - else conditiion, and you have only one IF, then return, and one else that return something default - I remove "else" condition because there it is unnecessary 

If there are more than one condition - it's true, always use else condition

f-meloni reviewed (changes requested):
Thank you for the PR :)
Looks good!
Just few small comments

Commit:
Added minor documentation in Epub
Added in Error - key localizedDescription
Returned return nil in else condition

Commit:
make epubContentParser internal again
added supported methods for Epub - content for Spine
removed allContent from epubContentParser

Commit:
a little bit improvements on EpubContentParser's - removed parentURL because it was the same as epubContentsURL

f-meloni commented:
Thank you :)
